KernelCommunicator.swift

```diff
@@ -59,8 +70,36 @@ public struct KernelCommunicator {
   }
 
   /// A single serialized display message for the Jupyter client.
+  ///
+  /// Refers to unmanaged memory so that the kernel can read the messages
+  /// directly from memory.
+  ///
+  /// Clients must call `deinitialize()` after they are finished with this
+  /// struct. The memory that this refers to becomes invalid when
+  /// `deinitialize()` is called.
+  ///
+  /// TODO: We could make this a class and take advantage of Swift's reference
+  //  counting. But there are some obstacles to overcome first:
+  /// 1. It seems that LLDB never releases values that it has references to.
+  ///    So we need to fix that to avoid leaking memory.
+  /// 2. When I make this a class, LLDB can't read the contents of `parts`.
+  ///    LLDB says "<read memory from {ADDRESS} failed (0 of 8 bytes read)>".
   public struct JupyterDisplayMessage {
-    public let parts: [[CChar]]
+    let parts: [UnsafeMutableBufferPointer<CChar>]
```

Comment:
It shouldn't need unsafe code, esp. `UnsafeMutableBufferPointer`, to improve efficiency here. Unsafe code makes it significantly harder to maintain. Let's chat in person.